docs/src/formula.md

```diff
@@ -14,167 +14,184 @@ fields with possibly heterogeneous types.  One of the primary goals of
 `StatsModels` is to make it simpler to transform tabular data into matrix format
 suitable for statistical modeling.
 
-At the moment, "tabular data" means an `AbstractDataFrame`.  Ultimately, the
-goal is to support any tabular data format that adheres to a minimal API,
-**regardless of backend**.
-
-## The `Formula` type
-
-The basic conceptual tool for this is the `Formula`, which has a left side and a
-right side, separated by `~`. Formulas are constructed using the `@formula` macro:
-
-```jldoctest
-julia> @formula(y ~ 1 + a)
-Formula: y ~ 1 + a
+At the moment, "tabular data" means a
+[Tables.jl](https://github.com/JuliaData/Tables.jl) table, which will be
+materialized as a `Tables.ColumnTable` (a `NamedTuple` of column vectors).  Work
+on first-class support for streaming/row-oriented tables is ongoing.
+
+## The `@formula` DSL
+
+StatsModels implements the [`@formula`](@ref) domain-specific language (DSL) for
+describing table-to-matrix transformations.  This DSL is designed to be familiar
+to users of other statistical software, while also taking advantage of Julia's
+unique strengths to be fast and flexible.
+
+A basic formula is composed of individual *terms*—symbols which refer to data
+columns, or literal numbers `0` or `1`—combined by `+`, `&`, `*`, and (at the
+top level) `~`.
+
+!!! note 
+
+    The `@formula` macro **must** be called with parentheses to ensure that
+    the formula is parsed properly.
+
+### An example
```

Comment:
Since that's the only heading at that level, I'd turn it into a normal sentence, e.g. "We now illustrate this with an example."